sql/src/main/java/io/crate/operation/projectors/ShardDMLExecutor.java

```diff
@@ -171,13 +157,15 @@ private String getLocalNodeId(ClusterService clusterService) {
         return nodeId;
     }
 
-    private void processShardResponse(ShardResponse shardResponse) {
-        if (shardResponse.failure() != null) {
-            executionFuture.completeExceptionally(shardResponse.failure());
-            return;
+    private BitSet processShardResponse(ShardResponse shardResponse) {
+        Exception failure = shardResponse.failure();
+        if (failure != null) {
+            Throwables.throwIfUnchecked(failure);
+            throw new RuntimeException(failure);
```

Comment:
In most cases it should be a RuntimeException anyway, and if it's not, it's probably not too bad to extend the stacktrace as the failure is a streamed exception and the stacktrace would be rather confusing if you try to figure out where the exception was raised.

I also don't see a performance issue here? This is a failure case, per request. Not per row or something